src/test/java/mate/academy/dao/impl/UserDaoTest.java

```diff
@@ -0,0 +1,87 @@
+package mate.academy.dao.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.dao.RoleDao;
+import mate.academy.dao.UserDao;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class UserDaoTest extends AbstractTest {
+    private UserDao userDao;
+    private RoleDao roleDao;
+    private User bob;
+    private Role role;
+
+    @BeforeEach
+    void setUp() {
+        role = new Role(Role.RoleName.ADMIN);
+        userDao = new UserDaoImpl(getSessionFactory());
+        roleDao = new RoleDaoImpl(getSessionFactory());
+        roleDao.save(role);
+        Set<Role> roles = Set.of(role);
+        bob = getUser("Bob@gmail.com", "Qwerty!234", roles);
+    }
+
+    @Test
+    void save_validUser_Ok() {
+        User actualUser = userDao.save(bob);
+        assertNotNull(actualUser);
+        assertEquals(bob, actualUser);
+    }
+
+    @Test
+    void save_notPresent_notOk() {
+        assertThrows(RuntimeException.class, () -> {
+            userDao.save(null);
+        });
+    }
+
+    @Test
+    void findByEmail_validEmail_Ok() {
+        userDao.save(bob);
+        User actual = userDao.findByEmail(bob.getEmail()).get();
+        assertNotNull(actual);
+        assertEquals(bob.getEmail(), actual.getEmail());
+    }
+
+    @Test
+    void findByEmail_notPresent_notOk() {
+        Optional<User> optionalUser = userDao.findByEmail("Asd");
+        assertTrue(optionalUser.isEmpty());
+    }
```

Comment:
```suggestion
    void findByEmail_notPresent_Ok() {
```

we have not some exception here